### ISSUE TITLE
[Workflow] Integration of Workflow Engine

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery#3934]
#### Current Behavior
Meshery has a built-in patterns engine, capable of building DAGs.

#### Desired Behavior
Meshery will benefit from integration of a robust and dedicated workflow engine.

This issue is an epic.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

**Comment by MarcaWitt** [comment]
@leecalcote I would like apply for this project under the LFX . I have read the previous comments and the requirements you mentioned. Looking forward to it.

**Comment by wisdomekpotu** [comment]
hey @leecalcote, My name is Wisdom Ekpotu, I am a Third Year Computer Engineering Student from the University of Uyo, Nigeria. I saw this project in the LFX mentorship fall 2021 and I thought this project had some parallels with my skills. I would love to contribute to this project and I'm open to learning any new technologies required for the project.
Also, I just joined the official Slack channel to get more information about the project. I am really looking forward to contributing to this project!

**Comment by wisdomekpotu** [comment]
Also @leecalcote, i just noticed that the link to the Wireframes and designs for Meshery UI in Figma is not found
![image](https://user-images.githubusercontent.com/58194343/130323462-c2857206-1568-48bb-9e03-97e5cfea0549.png)

**Comment by wisdomekpotu** [comment]
OR i guess i dont have permissions

**Comment by alphaX86** [comment]
@wisdomekpotu if you're new to the community, then join us on Slack (slack.meshery.io) signal us, we'll provide a form for Newcomers and then we'd provide access to the files

**Comment by MarcaWitt** [comment]
@alphaX86 19 years old you‘re cool.  I'm 21 years old,It's nice to meet.

**Comment by wisdomekpotu** [comment]
@alphaX86 okay got it

**Comment by ashutosh887** [comment]
Hi @leecalcote 
I'm interested to work on this Feature/Issue
Please consider me for this!

**Comment by loyaltytrooper** [comment]
@leecalcote I would also like to work on this enhancement, if possible

**Comment by leecalcote** [comment]
Yes: Build Meshery Server, UI, Operator, MeshSync, a couple of Adapters, review the Meshery Architecture slide deck (fill in your [community member form](https://meshery.io/community/newcomers)), review the Messaging Framework and Operations Center design spec, then write down your thoughts about what it would mean to have a retry, rollback, ignore, continue action on failed workflows.

**Comment by loyaltytrooper** [comment]
When can I start on this?

**Comment by ashutosh887** [comment]
@leecalcote Please assign this Issue to me 😊✨
